server/data/data-file/src/main/java/org/apache/james/sieverepository/file/SieveFileRepository.java

```diff
@@ -199,25 +203,25 @@ public void haveSpace(Username username, ScriptName name, long size) throws Quot
 
     @Override
     public List<ScriptSummary> listScripts(Username username) throws StorageException {
-        File[] files = getUserDirectory(username).listFiles();
-        List<ScriptSummary> summaries = new ArrayList<>(files.length);
         File activeFile = null;
         try {
             activeFile = getActiveFile(username);
         } catch (ScriptNotFoundException ex) {
             // no op
         }
-        for (File file : files) {
-            if (!SYSTEM_FILES.contains(file.getName())) {
-                summaries.add(new ScriptSummary(new ScriptName(file.getName()), isActive(file, activeFile)));
-            }
-        }
-        return summaries;
+
+        Predicate<File> isActive = isActiveValidator(activeFile);
+        return Stream.of(Optional.ofNullable(getUserDirectory(username).listFiles()).orElse(new File[]{}))
```

Comment:
no test?

Reply:
I should create a fs with wrong credentials, it looks overrated